### PR TITLE
Transaction

### PR DESCRIPTION
**Pull request by gokul-khoros**
transaction method

### CODE REVIEW

File: Banking/src/main/java/com/example/Banking/Controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        return userRepository.save(user);
+    }
+
+    @DeleteMapping("/DeleteUser/{id}")
+    public void deleteUser(@PathVariable(value = "id") Long userId){
+        userRepository.deleteById(userId);
+    }
+
+    @PutMapping("/transaction/{id}")
+        public String transaction(@PathVariable(value = "id") Long userId, @RequestBody User userAmount) throws ResourceNotFoundException {
```
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/Controller/UserController.java]: By userAmount you mean receiver?

Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/Controller/UserController.java]: userId is sender's userId?

File: Banking/src/main/java/com/example/Banking/Controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        User sender = userRepository.findById(userId)
+                .orElseThrow(() -> new ResourceNotFoundException("User not found for this id :: " + userId));
+
+        //deducting
+        if( sender.getBalance() > userAmount.getBalance()) {
+            long hi = sender.setBalance(sender.getBalance() - userAmount.getBalance());
```
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/Controller/UserController.java]: What does hi represent?

File: Banking/src/main/java/com/example/Banking/Controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            sender.setBalance(hi);
+            userRepository.save(sender);
+        }
+        else {
+            return "invalid balance";
```
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/Controller/UserController.java]: Avoid sending plain strings to the caller. Return error codes with error description to the caller. You can create your custom pojo for that purpose or you can use javax.ws.rs.core.Response

File: Banking/src/main/java/com/example/Banking/Controller/UserController.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        }
+        
+        //crediting
+        User reciever =userRepository.findById(userAmount.getId()).orElseThrow(() -> new ResourceNotFoundException("User not found for this id :: " + userId));
```
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/Controller/UserController.java]: There should be space before and after =

File: Banking/src/main/java/com/example/Banking/Controller/UserController.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        long hii = reciever.setBalance(reciever.getBalance() + userAmount.getBalance());
```
Comment (grewaviv) [inline Banking/src/main/java/com/example/Banking/Controller/UserController.java]: What does hii represents?

**Review by grewaviv — CHANGES_REQUESTED** [review]
Please also check the comments in https://github.com/khoros-campus-hires/BankingSystem/pull/2

**Comment by grewaviv** [comment]
Add .gitignore to avoid committing intellij settings or local config files

File: Banking/src/main/java/com/example/Banking/Controller/UserController.java (hunk 1 above)
Comment (gokul-khoros) [reply Banking/src/main/java/com/example/Banking/Controller/UserController.java]: userAmount represents the Sender. I will change it into senderDetails.

Comment (gokul-khoros) [reply Banking/src/main/java/com/example/Banking/Controller/UserController.java]: Yes, userid is sender's id.

File: Banking/src/main/java/com/example/Banking/Controller/UserController.java (hunk 2 above)
Comment (gokul-khoros) [reply Banking/src/main/java/com/example/Banking/Controller/UserController.java]: I'm sorry. It actually indicates the sender's new balance after the transaction.

File: Banking/src/main/java/com/example/Banking/Controller/UserController.java (hunk 1 above)
Comment (grewaviv) [reply Banking/src/main/java/com/example/Banking/Controller/UserController.java]: Why the transaction is based on userId? It should be on from and to accounts. A user can have multiple accounts. Also better create a separate pojo which represents a transaction having fields like fromAccount, toAccount and amount to represent a transaction

File: Banking/src/main/java/com/example/Banking/Controller/UserController.java (hunk 4 above)
Comment (gokul-khoros) [reply Banking/src/main/java/com/example/Banking/Controller/UserController.java]: Sure. I will correct this.

File: Banking/src/main/java/com/example/Banking/Controller/UserController.java (hunk 5 above)
Comment (gokul-khoros) [reply Banking/src/main/java/com/example/Banking/Controller/UserController.java]: I will change this too.

**Pull request closed without merging**